src/main/java/core/basesyntax/data/ShopService.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.data;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.util.FileWriter;
+import core.basesyntax.util.FileWriterImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopService {
```

Comment:
The `ShopService` class handles multiple responsibilities, such as processing transactions and generating reports. Consider separating these responsibilities into distinct services to adhere to the Single Responsibility Principle.